morpheus/models/dfencoder/autoencoder.py

```diff
@@ -737,251 +741,132 @@ def _create_stat_dict(self, a):
-                use_val_for_loss_stats is set to True.")
+    def _build_or_configure_datasets(self, train_data, val_data, use_val_for_loss_stats):
+        # Construct validation and trianing datasets or modify the DataLoader's dataset and return it
+        train_data = self._data_to_dset(train_data)
+        val_data = self._data_to_dset(val_data, train=False)
 
-        if use_val_for_loss_stats:
-            df_for_loss_stats = val.copy()
+        if (use_val_for_loss_stats):
+            loss_dset = val_data
         else:
-            # use train loss
-            df_for_loss_stats = df.copy()
+            loss_dset = train_data
 
-        if run_validation and val is not None:
-            val = val.copy()
+        if isinstance(loss_dset, torch.utils.data.DataLoader):
+            loss_dset = train_data.dataset
 
-        if self.optim is None:
-            self._build_model(df)
+        return train_data, val_data, loss_dset
 
-        if self.n_megabatches == 1:
-            df = self.prepare_df(df)
-
-        if run_validation and val is not None:
-            val_df = self.prepare_df(val)
-            val_in = val_df.swap(likelihood=self.swap_p)
-            msg = "Validating during training.\n"
-            msg += "Computing baseline performance..."
-            baseline = self.compute_baseline_performance(val_in, val_df)
-            LOG.debug(msg)
+    def _train_for_epochs(self, train_data, val_data, epochs, rank, world_size, run_validation, is_main_process):
+        # Batched Training loop with early stopping
+        count_es = 0
+        last_val_loss = float('inf')
+        should_early_stop = False
+        for epoch in range(epochs):
+            LOG.debug(f'[Rank{rank}] training epoch {epoch + 1}...')
 
-        n_updates = len(df) // self.batch_size
-        if len(df) % self.batch_size > 0:
-            n_updates += 1
-        last_loss = 5000
+            # if we are using DistributedSampler, we have to tell it which epoch this is
+            if (self.distributed_training
+                    and isinstance(train_data.sampler, torch.utils.data.distributed.DistributedSampler)):
+                train_data.sampler.set_epoch(epoch)
 
-        count_es = 0
-        for i in range(epochs):
-            self.train()
+            train_loss_sum = 0
+            train_loss_count = 0
+            for data_batch in train_data:
+                loss = self._fit_batch(**data_batch['data'])
 
-            LOG.debug(f'training epoch {i + 1}...')
-            df = df.sample(frac=1.0)
-            df = EncoderDataFrame(df)
-            if self.n_megabatches > 1:
-                self.train_megabatch_epoch(n_updates, df)
-            else:
-                input_df = df.swap(likelihood=self.swap_p)
-                self.train_epoch(n_updates, input_df, df)
+                train_loss_count += 1
+                train_loss_sum += loss
 
-            if self.lr_decay is not None:
+            if (self.lr_decay is not None):
```

Comment:
It stands for 'learning rate' decay. I don't think it would be confusing from a DS perspective, but I also don't see any reason not to make it's function more clear, will update.